src/app/visual/checkbox/checkbox-visual.component.html

```diff
@@ -66,3 +66,70 @@
     [checked]="false">
   </sky-checkbox>
 </div>
+<div>
+  <h3>
+    Template-driven checkbox
+  </h3>
+  <p>
+    <sky-checkbox
+      required
```

Comment:
Should this be an input just to make it easier to explain in the docs? I know it "works" because Angular looks at the native HTML attribute, but since this is a custom component, I feel that every attribute should be either an input or an output. Angular Material's checkbox component uses an input as well: https://material.angular.io/components/checkbox/api#MatCheckbox

Reply:
Hm... In a reactive form scenario, you'd be able to set a `[required]` input *and* add a `Validators.required` to the form control. Does that seem too confusing? Or better yet - what if you set one and not the other? Which one takes precedence?

Reply:
If you made it an input, Angular would throw a warning in a reactive form setup, so that would address that particular issue. In a template-driven setup, it would be expected for the user to set the required input. The docs could be something around the lines of, "Set the required attribute when using this component in a template-driven form. Use `Validators.required` on the form control itself in reactive forms." Again, since Angular throws a warning in this scenario, the developer should be able to fix this without referring to the docs every time.)

In a reactive setup when both the validators and the input are set: you could write some logic that only accepts the input value if the component is used in a template-driven setup. It's currently possible to check if a `ControlValueAccessor` is used in a reactive setup?

Reply:
After discussion, I added the `@Input()`.

Reply:
Should I create a docs issue to make sure this gets documented?